src/main.jsx

```diff
@@ -116,21 +116,20 @@ function App() {
     const [flip, setFlip] = useState(false)
     const pane = useRef()
 
-    function switchToTab(index, pushState) {
+    function switchToTab(index, pushState, prevTabIndex) {
         if (index === paneIndex || index === tabIndex) return
-
-        console.info("switching to tab", index)
-
         setRotation(rotation - 180)
-        setTabIndex(index)
+
+        console.info("switching to tab", index, "from ", prevTabIndex)
 
         if (pushState)
             history.pushState(null, tabs[index].name, tabs[index].slug)
 
         setTimeout(() => {
+            setTabIndex(index)
```

Comment:
moved this to mid rotation so that it doesn't show the previous tab content getting flipped. This is why info -> progress was a little glitchy.